topostats/tracing/skeletonize.py

```diff
@@ -2,78 +2,641 @@
+        on both its local binary environment.
+
+        Parameters
+        ----------
+        point: list
+            List of [x, y] coordinate positions
+
+        Returns
+        -------
+        bool
+            Returns T/F depending if the surrounding points have met the criteria
+            of the binary thin a, b returncount, c and d checks below.
+        """
+
+        self.p7, self.p8, self.p9, self.p6, self.p2, self.p5, self.p4, self.p3 = self.getLocalPixelsBinary(
+            self.mask, point[0], point[1]
+        )
+        return (
+            self._binaryThinCheck_a()
+            and self._binaryThinCheck_b_returncount() == 1
+            and self._binaryThinCheck_c()
+            and self._binaryThinCheck_d()
+        )
+
+    def _deletePixelSubit2(self, point):
+        """Function to check whether a single point should be deleted based
+        on both its local binary environment.
+
+        Parameters
+        ----------
+        point: list
+            List of [x, y] coordinate positions
+
+        Returns
+        -------
+        bool
+            Returns T/F depending if the surrounding points have met the criteria
+            of the binary thin a, b returncount, csharp and dsharp checks below.
+        """
+
+        self.p7, self.p8, self.p9, self.p6, self.p2, self.p5, self.p4, self.p3 = self.getLocalPixelsBinary(
+            self.mask, point[0], point[1]
+        )
+        # Add in generic code here to protect high points from being deleted
+        return (
+            self._binaryThinCheck_a()
+            and self._binaryThinCheck_b_returncount() == 1
+            and self._binaryThinCheck_csharp()
+            and self._binaryThinCheck_dsharp()
+        )
+
+    def _binaryThinCheck_a(self):
+        """Checks the surrounding area to see if the point lies on the edge of the grain.
+        Condition A protects the endpoints (which will be > 2)
+
+        Returns
+        -------
+        bool
+            if point lies on edge of graph and isn't an endpoint.
+        """
+        return 2 <= self.p2 + self.p3 + self.p4 + self.p5 + self.p6 + self.p7 + self.p8 + self.p9 <= 6
+
+    def _binaryThinCheck_b_returncount(self):
+        """Assess local area connectivity?"""
+        count = sum(
+            [
+                [self.p2, self.p3] == [0, 1],
+                [self.p3, self.p4] == [0, 1],
+                [self.p4, self.p5] == [0, 1],
+                [self.p5, self.p6] == [0, 1],
+                [self.p6, self.p7] == [0, 1],
+                [self.p7, self.p8] == [0, 1],
+                [self.p8, self.p9] == [0, 1],
+                [self.p9, self.p2] == [0, 1],
+            ]
+        )
+
+        return count
+
+    def _binaryThinCheck_c(self):
+        """Check if p2, p4 or p6 is 0 - seems very specific
+
+        Returns
+        -------
+        bool
+            if p2, p4 or p6 is 0.
+        """
+        return self.p2 * self.p4 * self.p6 == 0
+
+    def _binaryThinCheck_d(self):
```

Comment:
Ditto why this check?

Also this and some of the others might be combined. The products being calculated will be true if any one of the values is `0` (since `0 * anything = 0` and so a simpler check might be to ensure `p2` / `p4` / `p6` / `p8` are all non-zero as the tests are as they stand not mutually exclusive.

A better understanding of why this is required or considered a useful check would help, maybe @JoeBeton could help, or perhaps a close reading of the cited paper.